### ISSUE TITLE
Unsubscribing from AsyncEvent does not work.

### ISSUE THREAD

**Issue by Stannieman** [dotnet/MQTTnet#1479]
### Describe the bug
When unsubscribing from an AsyncEvent nothing is unsubscribed.

### Which component is your bug related to?

- AsyncEvent

### To Reproduce
Run the code below.
Every time MyFunc is executed the number of handlers in _myEvent increases.

### Expected behavior
When unsubscribing from an event the handler is removed from _myEvent.
### Code example
```csharp
private AsyncEvent<EventArgs> _myEvent = new();

public event Func<EventArgs, Task> MyEventAsync
{
    add => _myEvent.AddHandler(value);
    remove => _myEvent.RemoveHandler(value);
}

public MyFunc()
{
    MyEventAsync += Handler;
    MyEventAsync -= Handler;
}

private Task Handler(EventArgs arg)
{
    throw new NotImplementedException();
}
```

**Comment by chkr1011** [comment]
Thank you for reporting this issue. It will be fixed in version 4.1.

**Comment by sonali92jain** [comment]
> Thank you for reporting this issue. It will be fixed in version 4.1.
Hi, I want to check if this bug has been resolved yet. 
It was stated before closing that it would be fixed in version 4.1. I am currently using version 4.2.0.706 and the issue still persists

**Comment by chkr1011** [comment]
There are Unit Tests for that (see class AsyncEvent_Tests). Please let me know which scenario is not covered.
